src/coreclr/jit/codegenloongarch64.cpp

```diff
@@ -0,0 +1,9260 @@
+//    on the simdNode.
+//    Regarding spill, please see the note above on genSIMDIntrinsicUpperSave.  If we have spilled
+//    an upper-half to the lclVar's home location, this node will be marked GTF_SPILLED.
+//
+void CodeGen::genSIMDIntrinsicUpperRestore(GenTreeSIMD* simdNode)
+{
+    assert(!"unimplemented on LOONGARCH yet");
+}
+
+//-----------------------------------------------------------------------------
+// genStoreIndTypeSIMD12: store indirect a TYP_SIMD12 (i.e. Vector3) to memory.
+// Since Vector3 is not a hardware supported write size, it is performed
+// as two writes: 8 byte followed by 4-byte.
+//
+// Arguments:
+//    treeNode - tree node that is attempting to store indirect
+//
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genStoreIndTypeSIMD12(GenTree* treeNode)
+{
+    assert(!"unimplemented on LOONGARCH yet");
+}
+
+//-----------------------------------------------------------------------------
+// genLoadIndTypeSIMD12: load indirect a TYP_SIMD12 (i.e. Vector3) value.
+// Since Vector3 is not a hardware supported write size, it is performed
+// as two loads: 8 byte followed by 4-byte.
+//
+// Arguments:
+//    treeNode - tree node of GT_IND
+//
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genLoadIndTypeSIMD12(GenTree* treeNode)
+{
+    assert(!"unimplemented on LOONGARCH yet");
+}
+
+//-----------------------------------------------------------------------------
+// genStoreLclTypeSIMD12: store a TYP_SIMD12 (i.e. Vector3) type field.
+// Since Vector3 is not a hardware supported write size, it is performed
+// as two stores: 8 byte followed by 4-byte.
+//
+// Arguments:
+//    treeNode - tree node that is attempting to store TYP_SIMD12 field
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genStoreLclTypeSIMD12(GenTree* treeNode)
+{
+    assert(!"unimplemented on LOONGARCH yet");
+}
+
+#endif // FEATURE_SIMD
+
+/*****************************************************************************
+ * Unit testing of the LOONGARCH64 emitter: generate a bunch of instructions into the prolog
+ * (it's as good a place as any), then use COMPlus_JitLateDisasm=* to see if the late
+ * disassembler thinks the instructions as the same as we do.
+ */
+
+// Uncomment "#define ALL_LOONGARCH64_EMITTER_UNIT_TESTS" to run all the unit tests here.
+// After adding a unit test, and verifying it works, put it under this #ifdef, so we don't see it run every time.
+//#define ALL_LOONGARCH64_EMITTER_UNIT_TESTS
+
+#if defined(DEBUG)
+void CodeGen::genLoongArch64EmitterUnitTests()
+{
+    if (!verbose)
+    {
+        return;
+    }
+
+    if (!compiler->opts.altJit)
+    {
+        // No point doing this in a "real" JIT.
+        return;
+    }
+
+    // Mark the "fake" instructions in the output.
+    printf("*************** In genLoongArch64EmitterUnitTests()\n");
+
+    printf("*************** End of genLoongArch64EmitterUnitTests()\n");
```

Comment:
You don't have any emitter "unit tests" here. How did you test the instruction encodings generated by the JIT?

For our other architectures, we generate lots of different instructions using this mechanism, look at the JIT-generated disassembly, and compare the disassembly of these same instruction bytes by the debugger disassembler (typically windbg on Windows, lldb or gdb on Linux).

Reply:
You are  right.
Because these we had tested by all the coreclr-tests cases and many projects.
And also with the gdb.

So we didn't add this function to this PR.